rfd/0073-discover.md

```diff
@@ -0,0 +1,203 @@
+
+Users should be able to navigate between the dialog steps back and forth to
+connect multiple resource if needed. The unified "Add resource" wizard should
+also be available and prominently visible in the web UI allowing users to go
+through the same flow in a non-empty cluster.
+
+### Step 1. Gather information
+
+Ask the user what type of resource they would like to connect: an SSH server,
+a database, an application, etc.
+
+Then ask the user where the resource is located: self-hosted or AWS. For
+databases, also present the supported protocol options. For self-hosted:
+PostgreSQL, MySQL, MongoDB, Redis, SQL Server. For AWS: RDS PostgreSQL, RDS
+MySQL, RDS SQL Server, Elasticache, MemoryDB.
+
+### Step 2. Install Teleport
+
+For SSH, instruct the user to download and install `teleport` binary on the
+server they're intending to connect. Use tabs to display per-distro install
+instructions similar to [the docs](https://goteleport.com/docs/server-access/getting-started/#step-14-install-teleport-on-your-linux-host).
+
+For a database, display the same instructions making it clear that Teleport
+should be installed on a node that can reach the database.
+
+### Step 3. Configure node
+
+For SSH, this step is a no-op as it doesn't need any additional configuration
+so users will proceed to starting an agent at step 4.
+
+For databases, this step will inform the user of any additional configuration
+they may need in order to get database access to work. It will depend on the
+database type and where the database is hosted.
+
+For self-hosted databases, in the initial version the wizard will display the
+links to the respective sections of the documentation guides for preparing
+the node (e.g. joining to AD domain for SQL Server) and show the commands for
+creating a database user.
+
+For AWS databases that use IAM authentication (PostgreSQL, MySQL), the wizard
+will additionally display an appropriate `teleport db configure bootstrap`
+command for the user to run which will configure IAM permissions.
+
+### Step 4. Start agent
+
+For SSH, display the command for the user to run on their node, showing either
+"Automatic" or "AWS" commands from the existing "Add server" dialog depending
+on whether self-hosted or AWS option was picked in step 1.
+
+For a database, display the appropriate `teleport db start` command similar to
+the existing "Add database" dialog.
+
+### Step 5. Configure role
+
+Teleport roles by default only include internal user traits as allowed SSH
+logins (`{{internal.logins}}`) and database users (`{{internal.db_users}}`)
+and database names (`{{internal.db_names}}`). This results into users getting
+access denied errors unless they update their roles explicitly.
+
+On this step the wizard will ask the user for their intended SSH logins and/or
+database users/names and update the internal user traits appropriately so the
+users with the built-in `access` role will be allowed access to their resources.
+
+### Step 6. Connect
```

Comment:
Let's invite user to install Teleport connect here. It offers a much better US for getting started, especially for databases